src/machine/board_microbit.go

```diff
@@ -30,3 +35,60 @@ const (
 	SDA_PIN = 20
 	SCL_PIN = 19
 )
+
+// LED matrix pins
+const (
+	LED_COL_1 = 4
+	LED_COL_2 = 5
+	LED_COL_3 = 6
+	LED_COL_4 = 7
+	LED_COL_5 = 8
+	LED_COL_6 = 9
+	LED_COL_7 = 10
+	LED_COL_8 = 11
+	LED_COL_9 = 12
+	LED_ROW_1 = 13
+	LED_ROW_2 = 14
+	LED_ROW_3 = 15
+)
+
+// matrixSettings has the legs of the LED grid in the form {row, column} for each LED position.
+var matrixSettings = [5][5][2]uint8{
+	{{LED_ROW_1, LED_COL_1}, {LED_ROW_2, LED_COL_4}, {LED_ROW_1, LED_COL_2}, {LED_ROW_2, LED_COL_5}, {LED_ROW_1, LED_COL_3}},
+	{{LED_ROW_3, LED_COL_4}, {LED_ROW_3, LED_COL_5}, {LED_ROW_3, LED_COL_6}, {LED_ROW_3, LED_COL_7}, {LED_ROW_3, LED_COL_8}},
+	{{LED_ROW_2, LED_COL_2}, {LED_ROW_1, LED_COL_9}, {LED_ROW_2, LED_COL_3}, {LED_ROW_3, LED_COL_9}, {LED_ROW_2, LED_COL_1}},
+	{{LED_ROW_1, LED_COL_8}, {LED_ROW_1, LED_COL_7}, {LED_ROW_1, LED_COL_6}, {LED_ROW_1, LED_COL_5}, {LED_ROW_1, LED_COL_4}},
+	{{LED_ROW_3, LED_COL_3}, {LED_ROW_2, LED_COL_7}, {LED_ROW_3, LED_COL_1}, {LED_ROW_2, LED_COL_6}, {LED_ROW_3, LED_COL_2}}}
+
+// InitLEDMatrix initializes the LED matrix, by setting all of the row/col pins to output
+// then calling ClearLEDMatrix.
+func InitLEDMatrix() {
+	for i := LED_COL_1; i <= LED_ROW_3; i++ {
+		nrf.GPIO.DIRSET = 1 << uint8(i)
+	}
+	ClearLEDMatrix()
+}
+
+// ClearLEDMatrix clears the entire LED matrix.
+func ClearLEDMatrix() {
+	for i := LED_COL_1; i <= LED_COL_9; i++ {
+		nrf.GPIO.OUTSET = 1 << uint8(i)
+	}
```

Comment:
This produces smaller code (32 bytes less) and is probably faster as well:

```go
// ClearLEDMatrix clears the entire LED matrix.
func ClearLEDMatrix() {
        set := 0
        for i := LED_COL_1; i <= LED_COL_9; i++ {
                set |= 1 << uint8(i)
        }
        nrf.GPIO.OUTSET = nrf.RegValue(set)
        nrf.GPIO.OUTCLR = (1 << LED_ROW_1) | (1 << LED_ROW_2) | (1 << LED_ROW_3)
}
```

The reason, again, is that direct register access is expensive and with this change the whole `for` loop can be optimized away.

Reply:
Good to know. Change pushed.